Json serialization

jocelyn opened the pull request.

schoelle commented:
As usual, Microsoft does not care that an API has been deprecated more than 10 years. Well, such is life.

My suggestion would be to implement C_DATE using gettimeofday and to have use a compatibility library for Windows. The PostgreSQL people seem to have already done the work for us at, see http://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/port/gettimeofday.c;h=75a91993b74414c0a1c13a2a09ce739cb8aa8a08;hb=HEAD

Would that be a solution? I don't know if other OSes will drop 'ftime' any time soon, but it is definitely cleaner not to use a deprecated function call.

manuseiffel commented:
Could you find a way to implement both Windows/Linux/BSD/MacOSX?

jocelyn committed:
First implementation of JSON serialization and deserialization.
      It is still work in progress.
      Todo: complete the support for JSON array and object during deserialization.

jocelyn committed:
Updated json serialization to remove the specific reference support.

jocelyn committed:
Separated serialization with reference from the one without.

jocelyn committed:
Extracted implementation supporting references and cycle.
Made context required now on serializer and deserializer.
Better handling of error during deserialization.
Updated tests.

manuseiffel commented:
Closing. Merged into rev#99541
